examples/private_set_intersection/module/rust/src/lib.rs

```diff
@@ -46,27 +45,51 @@ oak::entrypoint!(oak_main => |_in_channel| {
     oak::run_event_loop(dispatcher, grpc_channel);
 });
 
+/// Maximum number of contributed private sets.
+pub const CONTRIBUTION_THRESHOLD: u64 = 2;
+
 #[derive(Default)]
 struct Node {
+    /// Current intersection of contributed private sets.
     values: Option<HashSet<String>>,
+    /// Number of contributed private sets.
+    set_count: u64,
+    /// Check whether the intersection is fixed and new contributions are discarded.
+    fixed: bool,
 }
 
 impl PrivateSetIntersection for Node {
     fn submit_set(&mut self, req: SubmitSetRequest) -> grpc::Result<()> {
-        let set = req.values.iter().cloned().collect::<HashSet<_>>();
-        let next = match self.values {
-            Some(ref previous) => previous.intersection(&set).cloned().collect(),
-            None => set,
-        };
-        self.values = Some(next);
-        Ok(())
+        if self.fixed {
+            return Err(grpc::build_status(
+                grpc::Code::InvalidArgument,
+                "Set contributions are no longer available",
+            ));
+        }
+
+        if self.set_count < CONTRIBUTION_THRESHOLD {
+            let set = req.values.iter().cloned().collect::<HashSet<_>>();
+            let next = match self.values {
+                Some(ref previous) => previous.intersection(&set).cloned().collect(),
+                None => set,
+            };
+            self.values = Some(next);
+            self.set_count += 1;
+            Ok(())
+        } else {
+            Err(grpc::build_status(
+                grpc::Code::InvalidArgument,
+                "Maximum number of contributed private sets is reached",
+            ))
+        }
     }
 
     fn get_intersection(&mut self, _req: ()) -> grpc::Result<GetIntersectionResponse> {
         let mut res = GetIntersectionResponse::default();
         if let Some(ref set) = self.values {
             res.values = set.iter().cloned().collect();
         };
+        self.fixed = true;
```

Comment:
This probably requires adding / updating the documentation on the protobuf service? I think callers would not expect that a call to a get method would cause side effects internally.